### PR TITLE
Prove every twin reset clears all persisted state

### PR DESCRIPTION
**Pull request by GaganSD**
Add a runtime database test that seeds, resets, and verifies every non-internal table is empty.

## What
Add a table-driven runtime test in `@pome-sh/sandbox-domains` that opens every first-party twin from its default seed, asserts the seed wrote persisted rows, invokes that twin's `resetDatabase` path, enumerates non-internal tables from `sqlite_master`, and asserts each is empty.

## Why
A digital twin is resettable. Reset must clear every table the seed actually created, not merely the tables a hand-maintained DELETE list remembers. F-1629.

## Notes for reviewer
No production reset SQL changed. The runtime check passed against the current lists for github, slack, stripe, gmail, and linear. The case table is keyed by `SANDBOX_DOMAIN_NAMES`, so a sixth twin without a harness is a type error.

### CODE REVIEW

File: packages/sandbox-domains/test/reset-clears-tables.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+function totalPersistedRows(db: Queryable, tables: string[]): number {
+  return tables.reduce((sum, table) => sum + tableRowCount(db, table), 0);
+}
+
+function assertResetClears(db: Queryable, reset: () => void, twin: string): void {
+  try {
+    const seededTables = persistedTables(db);
+    expect(seededTables.length, `${twin}: seed created no tables`).toBeGreaterThan(0);
+    expect(totalPersistedRows(db, seededTables), `${twin}: seed produced no rows`).toBeGreaterThan(0);
```
[the remark this answers is not in the document]
Comment (GaganSD) [reply packages/sandbox-domains/test/reset-clears-tables.test.ts:44]: Out of scope for this ticket. F-1629 asked for a seed-then-reset check against each twin's default seed: fail if that seed wrote no rows, then assert every non-internal sqlite_master table is empty. Filling every writable table would be a second, twin-specific world-construction suite, not the common runtime regression this PR adds.

Default-empty tables are still enumerated and asserted empty after reset. What this does not catch is a forgotten DELETE on a table the default seed never populated. That is a follow-up, not this change.

File: packages/sandbox-domains/test/reset-clears-tables.test.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    reset();
+    expect(
+      persistedTables(db).filter((table) => tableRowCount(db, table) > 0),
+      `${twin}: tables still populated after reset`,
+    ).toEqual([]);
```
[the remark this answers is not in the document]
Comment (GaganSD) [reply packages/sandbox-domains/test/reset-clears-tables.test.ts:50]: Good catch. The post-reset assertion now compares the sqlite_master table set to the post-seed set, then counts rows on that same list. A reset that drops a table fails the set comparison instead of disappearing from the emptiness check.
